p4include/v1model.p4

```diff
@@ -134,7 +265,36 @@ enum HashAlgorithm {
     xor16
 }
 
+/***
+ * mark_to_drop() is a primitive action that modifies
+ * standard_metadata.egress_spec to an implementation-specific special
```

Comment:
I would add that these changes happen after the pipeline has been completed. These do not happen when the operation is executed.

Reply:
I believe that is not true.  I believe that in the current p4c + simple_switch implementation, calls to mark_to_drop make modifications to standard_metadata.egress_spec and standard_metadata.mcast_grp _immediately_, which is why you can change those fields later and have the effect of the drop undone.

Reply:
It is recirculate/resubmit/clone operations that in P4_14 have a delayed effect, when they are requested to preserve metadata fields.  Even in P4_14, the drop action has an immediate effect that can later be changed before the end of ingress/egress processing, such that the drop can be undone.

Reply:
If that's true, this is another instance of magic action performed by an operation that does not refer to some fields, which should be impossible in p4-16. The compiler will assume that these fields cannot be changed by the mark_to_drop call. The only solution is to convert this call to something like `mark_to_drop(out standard_metadata meta)`;

Reply:
Wow, I can't believe I discovered all of this about recirculate/resubmit/clone, and didn't notice this :-)  You are right, mark_to_drop() is modifying something that it doesn't have explicit access to.

Reply:
I think you got the emoticon wrong, it should be :-(
Having this documentation early would have helped avoid this problem.
We should file another issue. We have to fix the p4-14 converter and invalidate a bunch of user programs.

Reply:
And I think it should be 'inout' direction if we make the entire standard_metadata struct a parameter to mark_to_drop(), because it should not change any of the many other fields inside that struct, except egress_spec and mcast_grp.

Adding such a parameter to mark_to_drop() is going to have an small effect on a lot of P4_16 test programs, I suspect, and other extant v1model code out there.  I am not saying that it should not be done -- just pointing out the potential impact.

Reply:
I can file a separate issue.

Would you prefer that I remove the part of these changes that document mark_to_drop?

Or keep them as they are, perhaps plus a note that mark_to_drop is violating the P4_16 language spec?

Reply:
We should eventually mark this as `@deprecated` and add the inout version.

Reply:
I have created the PR you described here: https://github.com/p4lang/p4c/issues/1828

Let me know if you want any further changes to this PR before approving it.

Reply:
@mbudiu-vmw No rush on this PR, since it is only documentation changes.  I just wanted to ensure you knew that unless you ask for other changes, or remind me of some you have asked for that I have forgotten, I am done changing this.